azurestorage_blueprints/move_file.py

```diff
@@ -0,0 +1,174 @@
+        dest='destination_folder_name',
+        default='',
+        required=False)
+    parser.add_argument(
+        '--destination-file-name',
+        dest='destination_file_name',
+        default=None,
+        required=False)
+    parser.add_argument(
+        '--connection-string',
+        dest='connection_string',
+        default=None,
+        required=True)
+    return parser.parse_args()
+
+
+def set_environment_variables(args):
+    """
+    Set Azure Connection String as environment variable if it's provided via keyword arguments
+    rather than seeded as environment variables. This will override system defaults.
+    """
+
+    if args.connection_string:
+        os.environ['AZURE_STORAGE_CONNECTION_STRING'] = args.connection_string
+    return
+
+
+def find_azure_storage_blob_file_names(conn_str, container_name, prefix=''):
+    """
+    Fetched all the files in the bucket which are returned in a list as
+    Azure Blob objects
+    """
+    container = ContainerClient.from_connection_string(
+        conn_str=conn_str, container_name=container_name, delimiter = '/')
+    blobs = list(container.list_blobs(prefix=prefix))
+    filtered = list(filter(lambda x: x.name.startswith(prefix),blobs))
+    return filtered
+        
+
+
+def find_matching_files(file_blobs, file_name_re):
+    """
+    Return a list of all file_names that matched the regular expression.
+    """
+    matching_file_names = []
+    for blob in file_blobs:
+        if re.search(file_name_re, blob.name):
+            matching_file_names.append(blob.name)
+
+    return matching_file_names
+
+
+def azure_move_blob(
+        connection_string,
+        container_name,
+        source_full_path,
+        destination_full_path):
+    """
+    Moves a single blob inside the same Azure Storage Blob Container.
+    Since there's no move funtion, this function is a combination of copy and delete
+    """
+    try:
+        blob_client = BlobServiceClient.from_connection_string(conn_str=connection_string,
+                                             container_name=container_name)
+    except:
+        print("Incorrect credentials")
+        sys.exit(ec.EXIT_CODE_INCORRECT_CREDENTIALS)
+    # get source blob url and copy to destination blob path
+    source_blob = blob_client.get_blob_client(container_name, source_full_path)
+    destination_blob = blob_client.get_blob_client(container_name, destination_full_path)
+
+    destination_blob.start_copy_from_url(source_blob.url, requires_sync=True)
+    copy_status_data = destination_blob.get_blob_properties().copy
+
+    if copy_status_data.status != "success":
+        # if copy failsm abort copy and return error message
+        destination_blob.abort_copy(copy_status_data.id)
+        print(f"Copy blob from {source_full_path} failed with status {copy_status_data.status}")
+        sys.exit(ec.EXIT_CODE_AZURE_MOVE_ERROR)
+    # successfully copied, so we delete the source path
+    source_blob.delete_blob()
+
```

Comment:
We need something in this function that prints a confirmation of the original path and the new path. Same thing as we had for S3/GCS

Reply:
Added in a message saying that the blob has been moved from `<original location>` to `<new location>`